pilot/bin/upload-istioctl

```diff
@@ -59,7 +59,6 @@ STATIC=0 GOOS=linux ${GOBUILD} "${BIN_DIR}/istioctl-linux" ${VERSION_PACKAGE} is
 
```

Comment:
Can we move this file (and similar in other parts) to release ? I would like to clean the pilot,mixer,securit/bin dirs.

Reply:
I'd also like to see the distributed directories go away, though it looks like istio/bin/push still relies on the file, and I think "make push" is still based on the file, so upload-istioctl isn't quite specific to the releases yet.  I would have tried to get rid of the release build's use of upload-istioctl but I wanted to first get opinions on acceptable approaches for that.  

I'm used to having arch-specific "bin" and/or "out" directories (e.g., bin/osx, bin/win, etc.), but if that's too annoying then they could possibly just be used when a non-native arch is specified.  Otherwise, we could try to have "go build" never create "istioctl" and instead always use a name from ("istioctl-linux", "istioctl-osx", "istioctl-win.exe") but offhand I don't know if there's a decent way to get "go build" to do that (and lots of people are probably in the habit of just using the same "istioctl", though I don't know if anyone is yet in the habit of running the tool out of the bin/ directory).

The last fallback I can think of is just to run "make istioctl" 3 times (with the appropriate options to build a different arch each time) and copy out the file between each run (possibly doing a clean depending on how much I/we want to trust "go build" to properly handle a change in arch), but it seemed a bit hacky and more fragile than other approaches.